store/src/java/com/zimbra/util/ExternalVolumeInfoHandler.java

```diff
@@ -192,4 +192,35 @@ public void modifyServerProperties(VolumeInfo volInfo) throws ServiceException,
             throw e;
         }
     }
+
+    /**
+     * Validate global bucket ID by reading LDAP atrribute
+     * @param globalS3BucketId
+     * @returns true if "global bucket ID" is valid
+     * @throws ServiceException, JSONException
+     */
+    public Boolean validateGlbBucketID(String globalS3BucketId) throws ServiceException, JSONException {
+        Boolean isValid = false;
+        try {
+            // step 1: Fetch current JSON state object and current JSON state array
+            String globalExternalStoreConfigJson = provisioning.getConfig().getGlobalExternalStoreConfig();
+            JSONObject currentJsonObject = new JSONObject(globalExternalStoreConfigJson);
+            JSONArray currentJsonArray = currentJsonObject.getJSONArray("global/s3BucketConfigurations");
+
+            // step 2: Find "globalBucketUUID" in current JSON array
+            for (int i = 0; i < currentJsonArray.length(); i++) {
+
+                // step 3: Mark validation as true if "globalBucketUUID" found
+                if (globalS3BucketId.equalsIgnoreCase(currentJsonArray.getJSONObject(i).getString("globalBucketUUID"))) {
```

Comment:
add constant for globalBucketUUID

Reply:
We are soon going to have new repo and constants file for common stuff. Will replace all constants when we add new repo